learn/sherif/main_test/src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -1,40 +1,48 @@
 package frc.robot.subsystems;
 
-import edu.wpi.first.wpilibj.SpeedController;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.drive.DifferentialDrive;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
-import java.math.BigDecimal;
-import java.math.RoundingMode;
-
 import static frc.robot.Constants.*;
 
 public class DriveTrain extends SubsystemBase {
-    private SpeedController left;
-    private SpeedController right;
+    private final WPI_TalonSRX m_left_a;
+    private final WPI_TalonSRX m_left_b;
+    private final WPI_TalonSRX m_left_c;
 
-    public DriveTrain(Factory f) {
-        this.left = f.getSparkMotor(FRONT_MOTOR);
-        this.right = f.getSparkMotor(BACK_MOTOR);
-        this.right.getInverted();
-    }
+    private final SpeedControllerGroup left;
 
-    public void drive(double ySpeed, double xSpeed) {
-        this.right.set((Clip(ySpeed + xSpeed)));
-        this.left.set((Clip(ySpeed - xSpeed)));
-    }
+    private final WPI_TalonSRX m_right_a;
+    private final WPI_TalonSRX m_right_b;
+    private final WPI_TalonSRX m_right_c;
 
-    public void brake() {
-        this.right.set(0.0);
-        this.left.set(0.0);
+    private final SpeedControllerGroup right;
+
+    private final DifferentialDrive differentialDrive;
+
+    public DriveTrain(Factory f) {
+        this.m_left_a = f.getTalonMotor(DRIVE_LEFT_MOTOR_A);
+        this.m_left_b = f.getTalonMotor(DRIVE_LEFT_MOTOR_B);
+        this.m_left_c = f.getTalonMotor(DRIVE_LEFT_MOTOR_C);
+        this.left = new SpeedControllerGroup(this.m_left_a, this.m_left_b, this.m_left_c);
+
+        this.m_right_a = f.getTalonMotor(DRIVE_RIGHT_MOTOR_A);
+        this.m_right_b = f.getTalonMotor(DRIVE_RIGHT_MOTOR_B);
+        this.m_right_c = f.getTalonMotor(DRIVE_RIGHT_MOTOR_C);
+        this.right = new SpeedControllerGroup(this.m_right_a, this.m_right_b, this.m_right_c);
+        this.right.setInverted(true);
```

Comment:
Why is this inverted?  I don't see any mention of needing to invert this in the documentation:
https://first.wpi.edu/FRC/roborio/beta/docs/java/edu/wpi/first/wpilibj/drive/DifferentialDrive.html

Reply:
From what my tests have seen, the differential drive code automatically inverts the input of one side if the motors themselves aren't inputted, providing essentially the same result with simpler function calls. As for why it needs to be inverted, from what I remember from FTC and I think should still apply to FRC, one side's motors are pointing in the opposite direction as the others, meaning one side would need to be inverted for both sides to drive in the same direction.